broker/tasks/route53.py

```diff
@@ -63,32 +63,31 @@ def remove_TXT_records(operation_id: int, **kwargs):
 
     for certificate in service_instance.certificates:
         for challenge in certificate.challenges:
-            domain = challenge.validation_domain
-            txt_record = f"{domain}.{config.DNS_ROOT_DOMAIN}"
-            contents = challenge.validation_contents
-            logger.info(f'Removing TXT record {txt_record} with contents "{contents}"')
-            try:
-                route53_response = route53.change_resource_record_sets(
-                    ChangeBatch={
-                        "Changes": [
-                            {
-                                "Action": "DELETE",
-                                "ResourceRecordSet": {
-                                    "Type": "TXT",
-                                    "Name": txt_record,
-                                    "ResourceRecords": [{"Value": f'"{contents}"'}],
-                                    "TTL": 60,
-                                },
-                            }
-                        ]
-                    },
-                    HostedZoneId=config.ROUTE53_ZONE_ID,
-                )
-            except:  # noqa E722
-                logger.info("Ignoring error because we don't care")
-            else:
-                change_id = route53_response["ChangeInfo"]["Id"]
-                logger.info(f"Ignoring Route53 TXT change ID: {change_id}")
+            _delete_TXT_record(challenge)
+
+
+@huey.retriable_task
+def remove_old_DNS_records(operation_id: int, **kwargs):
+    operation = db.session.get(Operation, operation_id)
+    service_instance = operation.service_instance
+
+    operation.step_description = "Removing old DNS records"
+    flag_modified(operation, "step_description")
+    db.session.add(operation)
+    db.session.commit()
+
+    # TODO: do we only need to look at the current certificate?
+    current_cert = service_instance.current_certificate
+    challenges = current_cert.challenges.all()
+
+    challenges_to_remove = [
+        challenge
+        for challenge in challenges
+        if challenge.domain not in service_instance.domain_names
+    ]
+    for challenge in challenges_to_remove:
+        _delete_TXT_record(challenge)
+        _delete_ALIAS_record(challenge.domain, service_instance)
```

Comment:
I decided to delete both the old TXT and ALIAS records in the same task because it's only possible to detect what ALIAS record(s) to delete using the same logic that is used to delete old TXT records